### ISSUE TITLE
Do not report CloudWatch custom metrics using SDK calls, use Embedded Format instead

### ISSUE THREAD

**Issue by RafalWilinski** [Stedi/eslint-plugin-stedi-aws-rules#2]
### Rule name

```
no-cloudwatch-direct-reporting
```

### Use case description

Reporting metrics using Cloudwatch SDK calls creates unnecessary overhead and makes your Lambdas slower. You should use [Embedded Metric Format](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html) to log any custom metrics inside Lambdas.

#### Incorrect

```js
var cloudwatch = new AWS.CloudWatch();
cloudwatch.getMetricWidgetImage(params, function (err, data) {
  if (err) console.log(err, err.stack); 
  else     console.log(data);
});
```

#### Correct

```js
const { metricScope } = require("aws-embedded-metrics");

const myFunc = metricScope(metrics =>
  async () => {
    // ...
  });

exports.handler = myFunc();
```

References:
- [https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html)
- [https://aws.amazon.com/about-aws/whats-new/2019/11/amazon-cloudwatch-launches-embedded-metric-format/](https://aws.amazon.com/about-aws/whats-new/2019/11/amazon-cloudwatch-launches-embedded-metric-format/)
- [https://github.com/awslabs/aws-embedded-metrics-node](https://github.com/awslabs/aws-embedded-metrics-node)
